cmd/erasure-zones.go

```diff
@@ -151,6 +151,33 @@ func (z *erasureZones) getZonesAvailableSpace(ctx context.Context) zonesAvailabl
 	return zones
 }
 
+// getZoneIdx returns the found previous object and its corresponding zone idx,
+// if none are found falls back to most available space zone.
+func (z *erasureZones) getZoneIdx(ctx context.Context, bucket, object string, opts ObjectOptions) (idx int, err error) {
+	for i, zone := range z.zones {
```

Comment:
We might as well make it safe/fast to use with a a single zone and not rely on the caller to check.
```suggestion
	if z.SingleZone() {
		return 0, nil
	}
	
	for i, zone := range z.zones {
```

Reply:
Good idea .